### ISSUE TITLE
Connection errors when downloading binary files from iOS & desktop Safari web browsers

### ISSUE THREAD

**Issue by efosao** [oven-sh/bun#7830]
### What version of Bun is running?

1.0.18+36c316a24

### What platform is your computer?

oven/bun:latest (docker)

### What steps can reproduce the bug?

1. Deploy a bun web server providing any static binary assets (jpg, png, zip etc) to a hosted server with https (I'm using the docker `oven/bun` docker container in dokku).
2. Trying to download those assets in Safari results in an error.
    - `Error: The network connection was lost.`
3. Error is reproduceable in `Elysia`, `Hono` or plain `Bun`. 
4. Error is **NOT** reproduceable running under `localhost` with no _https_.

![image](https://github.com/oven-sh/bun/assets/852788/c53f865c-8edb-4e60-aff1-f7a00c03d656)

### What is the expected behavior?

- Binary resources like images should load and be displayed properly
- Zip file links should be downloadable

<img width="1126" alt="image" src="https://github.com/oven-sh/bun/assets/852788/8aa90be5-cbed-455c-8efb-3167d162f6a3">

### What do you see instead?

- Binary resources like images do not display in Safari desktop and iOS
- Zip file links downloads error out and don't download all the way.
- However, binary images and links work in `Chrome` flawlessly.
- Binary images and links also work when Bun is replaced with Node and files are served in Express.

![image](https://github.com/oven-sh/bun/assets/852788/c53f865c-8edb-4e60-aff1-f7a00c03d656)

### Additional information

Hosted example link - https://bun.vauntly.com/about

- Image is broken on Safari Desktop Version 17.2.1 (19617.1.17.11.12)
- Image is broken on Safari iOS Version 17.1.2

```tsx
// code for web server
// Honon, Elysia & plain Bun implementations exhibited the same issues with Safari (mobile/desktop) binary downloads from /public folder
import { Hono } from 'hono'
import { serveStatic } from 'hono/bun'
import { cors } from 'hono/cors'

import pages from './controllers/pages'

const port = process.env.PORT || 80;
export const app = new Hono()

app
  .use(cors())
  .use('/public/*', serveStatic({ root: './src' }))
  .route('/', pages)

export default app
```

**Comment by Electroid** [comment]
It's possible that Safari and Chrome are sending slightly different HTTP requests (e.g. headers, protocol used, etc) and Bun is not handling the case for whatever Safari is doing. It would be helpful to see the request headers from Safari vs Chrome.

**Comment by efosao** [comment]
> It's possible that Safari and Chrome are sending slightly different HTTP requests (e.g. headers, protocol used, etc) and Bun is not handling the case for whatever Safari is doing. It would be helpful to see the request headers from Safari vs Chrome.

@Electroid Here you go. Also, I'm keeping the page live at https://bun.vauntly.com/about to assist with testing and verifying this bug.

## Chrome Headers
![image](https://github.com/oven-sh/bun/assets/852788/15cac4eb-eb6f-4ac0-9a0e-478cebd95263)

## Safari Headers
![image](https://github.com/oven-sh/bun/assets/852788/b57d0194-4b7d-490f-987e-13120da2646a)

**Comment by efosao** [comment]
@Electroid I have a both _Bun_ and _Node_ versions available to test here if that helps. Thanks for looking into this!! 👍🏽 

- Bun - https://bun.vauntly.com/about
- Node/Astro - https://astro.vauntly.com/about

**Comment by efosao** [comment]
@Electroid I believe this issue is being caused by the lack of HTTP/2 support in Bun. The Nginx Proxy Server is using HTTP/2. It's still curious why it runs in Chrome at all.

**Comment by efosao** [comment]
Issue was with Nginx translation/ graceful-downgrade failure between HTTP/1.1 and HTTP/2 (which Bun doesn't directly support) in version Nginx 1.18. The problem with streaming binary files was resolved by upgrading to Nginx 1.24.
